### PR TITLE
Add detailed VEX example Annex

### PR DESCRIPTION
**Pull request by rnjudge**
This commit adds an annex explaining how to implement VEX in SPDX.

### CODE REVIEW

**Comment by rnjudge** [comment]
cc @puerco @jeff-schutt to take a look

File: docs/annexes/how-to-implement-VEX-in-SPDX.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+For all VEX Relationships, the `from` element  must be a [Vulnerability](https://spdx.github.io/spdx-spec/v3.0/model/Security/Classes/Vulnerability/) and the `to` end of the relationship must point to one or more elements representing the VEX products. To specify a different element where the vulnerability was detected, the VEX relationship can optionally specify subcomponents using the [assessedElement](https://spdx.github.io/spdx-spec/v3.0/model/Security/Properties/assessedElement/) property.
+
+VEX inherits information from the document level down to its statements. When a statement is missing information it can be completed by reading the equivalent field from the containing document. For example, if a VEX relationship is missing data in its createdBy property, tools must consider the entity listed in the CreationInfo section of the document as the VEX author. In the same way, when a VEX relationship does not have a created property, the document's date must be considered as authoritative.
+
+
+## J.1 Assembling a VEX Statement
+
+A VEX statement is assembled by a triad of (at least):
+
+a software package + a vex assessment relationship + a vulnerability
```
Comment (puerco) [inline docs/annexes/how-to-implement-VEX-in-SPDX.md:20]: Here we are talking about VEX, so I would start with the vex lingo and then explain how each vex element is represented in SPDX:

```suggestion
A VEX statement is assembled by a triad of (at least):

a software product + a vex status + a vulnerability

In SPDX the VEX product can be any software package. There are four VEX status labels, each of which is represented in SPDX with a subclassed VEX Relationship:

| VEX Status | SPDX VEX Relationship |
| --- | --- |
| not_affected | [VexNotAffectedVulnAssessmentRelationship](https://spdx.github.io/spdx-spec/v3.0/model/Security/Classes/VexNotAffectedVulnAssessmentRelationship/) |
| affected | [VexAffectedVulnAssessmentRelationship](https://spdx.github.io/spdx-spec/v3.0/model/Security/Classes/VexAffectedVulnAssessmentRelationship/) |
| fixed | [VexFixedVulnAssessmentRelationship](https://spdx.github.io/spdx-spec/v3.0/model/Security/Classes/VexFixedVulnAssessmentRelationship/) |
| under_investigation | [VexUnderInvestigationVulnAssessmentRelationship](https://spdx.github.io/spdx-spec/v3.0/model/Security/Classes/VexUnderInvestigationVulnAssessmentRelationship/) | 

Also, feel free to steal [this ASCII art we have on openvex](https://github.com/openvex/spec/blob/main/OPENVEX-SPEC.md#the-vex-statement) or adapt it as we see fit.
```

**Review by puerco — COMMENTED** [review]
oooh thanks @rnjudge you beat me to starting this document. This is a good start!

I think we are missing an overview of how VEX works in general. I think it would be helpful to first get the idea of the mechanics into the reader's head and then building up on how SPDX implements them.

I'm happy to help with some of it :)

File: docs/annexes/how-to-implement-VEX-in-SPDX.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  "to": ["urn:product-acme-application-1.3"],
+  "assessedElement": "urn:npm-elliptic-6.5.2",
+  "suppliedBy": ["urn:spdx.dev:agent-jane-doe"],
+  "publishedTime": "2021-03-09T11:04:53Z"
+```
+
+
+## J.2 Changing the Status of a Vulnerability
+
+Because [Elements](https://spdx.github.io/spdx-spec/v3.0/model/Core/Classes/Element/) in SPDX are immutable, a new VEX Assessment Relationship of type `amends` must be issued each time the VEX status of a vulnerability changes (i.e. `underInvestigationFor` --> `affects`) in addition to creating a new type of VEX status relationship. The following example shows how you would communicate that a vulnerbaility was under investigation before determining that the vulnerability indeed affects a product.
```
[the remark this answers is not in the document]
Comment (rnjudge) [reply docs/annexes/how-to-implement-VEX-in-SPDX.md:38]: @puerco my understanding was that the amends relationship is how we string together evolving impact statements -- did this change?

**Comment by rnjudge** [comment]
> oooh thanks @rnjudge you beat me to starting this document. This is a good start!
> 
> I think we are missing an overview of how VEX works in general. I think it would be helpful to first get the idea of the mechanics into the reader's head and then building up on how SPDX implements them.
> 
> I'm happy to help with some of it :)

My assumption was that if folks are coming to this annex they are probably already familiar with how VEX works and I didn't want to clutter the page with too much background. But I'm not opposed to a brief primer! Maybe easier to point them to an already written summary of it somewhere for more detailed info?

Thanks so much for your help with the first pass, I'm incorporating your suggestions and we can keep iterating on this! Maybe review at the security all on Wed if you're around?

File: docs/annexes/how-to-implement-VEX-in-SPDX.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rnjudge) [reply docs/annexes/how-to-implement-VEX-in-SPDX.md]: @puerco -- @jeff-schutt and I discussed and agree that using an `amendedBy` relationship is the best way to preserve context and also make it easy for tools to search for updated VEX relationships. Of course, this is not a must and I've updated the verbiage in my PR to reflect that. Please continue to review and provide feedback at your convenience :)

**Comment by rjb4standards** [comment]
The SPDX V 3 spec says nothing about NIST's VDR - this is a significant oversight.

**Comment by Sintayew4** [comment]
#933

**Review by Sintayew4 — APPROVED** [review]
- [1fe2c09842d58d16de5f03503799861963bae84b ]

**Comment by zvr** [comment]
This should be moved to the https://github.com/spdx/using repository.

**Comment by rjb4standards** [comment]
The material shown in the commit matches my understanding of how VEX is used to communicate vulnerability status.

**Comment by goneall** [comment]
Reviewed on the 30 July 2024 tech call, approved.

We need to open this PR on the using repo and close this.  Once the PR is open on the using repo, we should be able to merge it as approved.

@rnjudge - can you move this over?

**Comment by kestewart** [comment]
THis has been moved over to "Using" so going ahead and closing this.

**Pull request closed without merging**